tuning/test_tune.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+from tune import (
+    Configuration, get_contract_tile_sizes, get_pipeline_config, get_shape_dims,
+    get_shapes_mmt, get_shapes_conv, get_shapes_contract, get_shapes_batch_matmul
+)
+
+'''
+Usage: python -m pytest test_tune.py
+'''
+
+def test_get_contract_tile_sizes():
+    config = Configuration(
+        subgroup_size=32,
+        workgroup_size=[16, 16, 1],
+        intrinsic="",
+        tile_sizes=[4, 8, 16],
```

Comment:
Can we try something more realistic with more than 3 dimensions? You can check the actual contracts here: https://github.com/nod-ai/sdxl-scripts/blob/37ef1bc8cc09d9a9afbf05e8faebcc4b331bf269/specs/attention_and_matmul_spec.mlir#L913

Reply:
m,n,k values are already set before Configuration passed to this function, we should cover this test by adding other function